api/apps/v1alpha1/nemo_datastore_types.go

```diff
@@ -157,99 +203,306 @@ func (n *NemoDatastore) GetStandardLabels() map[string]string {
+					Key: "jwtSecret",
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: n.Spec.Secrets.LfsJwtSecret,
+					},
+				},
+			},
+		},
 		{
-			ConfigMapRef: &corev1.ConfigMapEnvSource{
-				LocalObjectReference: corev1.LocalObjectReference{
-					Name: n.Spec.DataStoreParams.EnvConfigMap,
+			Name: "GITEA__DATABASE__PASSWD",
+			ValueFrom: &corev1.EnvVarSource{
+				SecretKeyRef: &corev1.SecretKeySelector{
+					Key: dbSetting.Credentials.PasswordKey,
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: dbSetting.Credentials.SecretName,
+					},
 				},
 			},
 		},
 		{
-			SecretRef: &corev1.SecretEnvSource{
-				LocalObjectReference: corev1.LocalObjectReference{
-					Name: n.Spec.DataStoreParams.EnvSecret,
+			Name: "GITEA_ADMIN_USERNAME",
+			ValueFrom: &corev1.EnvVarSource{
+				SecretKeyRef: &corev1.SecretKeySelector{
+					Key: "GITEA_ADMIN_USERNAME",
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: n.Spec.Secrets.GiteaAdminSecret,
+					},
 				},
 			},
 		},
+		{
+			Name: "GITEA_ADMIN_PASSWORD",
+			ValueFrom: &corev1.EnvVarSource{
+				SecretKeyRef: &corev1.SecretKeySelector{
+					Key: "GITEA_ADMIN_PASSWORD",
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: n.Spec.Secrets.GiteaAdminSecret,
+					},
+				},
+			},
+		},
+		{
+			Name:  "GITEA__LFS__SERVE_DIRECT",
+			Value: strconv.FormatBool(objStoreSetting.ServeDirect),
+		},
+		{
+			Name:  "GITEA__LFS__STORAGE_TYPE",
+			Value: "minio",
+		},
+		{
+			Name:  "GITEA__LFS__MINIO_ENDPOINT",
+			Value: objStoreSetting.Endpoint,
+		},
+		{
+			Name:  "GITEA__LFS__MINIO_BUCKET",
+			Value: objStoreSetting.BucketName,
+		},
+		{
+			Name:  "GITEA__LFS__MINIO_LOCATION",
+			Value: objStoreSetting.Region,
+		},
+		{
+			Name:  "GITEA__LFS__MINIO_LOCATION",
+			Value: objStoreSetting.Region,
+		},
+		{
+			Name:  "GITEA__LFS__MINIO_USE_SSL",
+			Value: strconv.FormatBool(objStoreSetting.SSL),
+		},
+		{
+			Name:  "GITEA__DATABASE__SSL_MODE",
+			Value: "disable",
+		},
+		{
+			Name:  "GITEA__DATABASE__NAME",
+			Value: dbSetting.DatabaseName,
+		},
+		{
+			Name:  "GITEA__DATABASE__HOST",
+			Value: fmt.Sprintf("%s:%d", dbSetting.Host, dbSetting.Port),
+		},
+		{
+			Name:  "GITEA__DATABASE__USER",
+			Value: dbSetting.Credentials.User,
+		},
 	}
+	return envVars
 }
 
-func (n *NemoDatastore) GetInitContainers() []corev1.Container {
-	image := n.Spec.DataStoreParams.InitContainerImage
-	if image == "" {
-		image = "busybox"
-	}
-	cmd := n.Spec.DataStoreParams.InitContainerCommand
-	if len(cmd) == 0 {
-		cmd = []string{
-			"sh",
-			"-c",
-			fmt.Sprintf("until nc -z %s %s; do echo \"PostgreSQL is unavailable. Sleeping for 5 seconds\"; sleep 5; done;", n.Spec.DataStoreParams.DatabaseHost, n.Spec.DataStoreParams.DatabasePort),
-		}
-	}
-	return []corev1.Container{
+// GetVolumes returns volumes for the NemoDatastore container
+func (n *NemoDatastore) GetVolumes() []corev1.Volume {
+	/*volumes:
```

Comment:
Is this comment needed?

Reply:
It helps folks to see what the generated volume would look like in the deployment manifest IMO